src/psyclone/psyGen.py

```diff
@@ -3111,15 +3120,20 @@ def step_expr(self, expr):
         :param expr: New PSyIR step expression.
         :type expr: :py:class:`psyclone.psyGen.Node`
 
+        :raises TypeError: if expr is not a PSyIR node.
+
         '''
+        if not isinstance(expr, Node):
+            raise TypeError(
+                "Only PSyIR nodes can be assigned as the Loop step expression"
+                ", but found '{0}' instead".format(type(expr)))
         self._check_completeness()
         self._children[2] = expr
 
     @property
     def loop_body(self):
-        ''' Return PSyIR Schedule with the loop body statements.
-
-        :return: PSyIR Schedule with the loop body statements
+        '''
+        :return: the PSyIR Schedule with the loop body statements.
         :rtype: list of :py:class:`psyclone.psyGen.Schedule`
```

Comment:
Not an error from this PR but presumably we don't really return a `list` of `Schedule`s do we?

Reply:
Good catch